### PR TITLE
feat(stream-model-instance,stream-model-instance-handler): Enable deterministic MIDs for models with SINGLE accountRelations

### PR DESCRIPTION
**Pull request by stbrody**
CC @PaulLeCam

### CODE REVIEW

File: packages/stream-model-instance/src/model-instance-document.ts [hunk 1]
```diff
@@ -120,6 +132,29 @@ export class ModelInstanceDocument<T = Record<string, any>> extends Stream {
     )
   }
 
+  /**
+   * Creates a deterministic Tile document.
+   * @param ceramic - Instance of CeramicAPI used to communicate with the Ceramic network
+   * @param metadata - Genesis metadata
+   * @param opts - Additional options
+   */
+  static async deterministic<T>(
```
Comment (stbrody) [inline packages/stream-model-instance/src/model-instance-document.ts:141]: I also considered calling this method `single()`, curious what you think about that @PaulLeCam

File: packages/common/src/utils/stream-utils.ts [hunk 2]
```diff
@@ -62,7 +63,7 @@ export class StreamUtils {
     }
 
     if (commit.header?.model) {
-      cloned.header.model = commit.header.model.toString()
+      cloned.header.model = uint8arrays.toString(commit.header.model, 'base64')
```
[the remark this answers is not in the document]
Comment (PaulLeCam) [reply packages/common/src/utils/stream-utils.ts:66]: What does it mean concretely on the consuming side, do we always deal with a `Uint8Array`, a `string` or it depends?
If a string, could we use something else that `base64` encoding please? Anything that's URL-safe so it doesn't become an issue in Web-based tools.

File: packages/stream-model-instance/src/model-instance-document.ts (hunk 1 above)
Comment (PaulLeCam) [reply packages/stream-model-instance/src/model-instance-document.ts:141]: I like `single()`, should be more explicit considering that's the term we use for the relation, but I don't feel strongly about it so no problem keeping as `deterministic()`.

**Review by willex — APPROVED** [review]

File: packages/common/src/utils/stream-utils.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (stbrody) [reply packages/common/src/utils/stream-utils.ts:66]: @oed any thoughts on how this should be serialized?

[the remark this answers is not in the document]
Comment (stbrody) [reply packages/common/src/utils/stream-utils.ts:66]: > What does it mean concretely on the consuming side, do we always deal with a Uint8Array, a string or it depends?

Apps layers shouldn't have to worry about how this is encoded.  The Ceramic http client already handles deserializing this data back to the expected types.

[the remark this answers is not in the document]
Comment (oed) [reply packages/common/src/utils/stream-utils.ts:66]: This seems fine. We are not going to encode the entire json-state in a URL anyway.

[the remark this answers is not in the document]
Comment (PaulLeCam) [reply packages/common/src/utils/stream-utils.ts:66]: > > What does it mean concretely on the consuming side, do we always deal with a Uint8Array, a string or it depends?
> 
> Apps layers shouldn't have to worry about how this is encoded. The Ceramic http client already handles deserializing this data back to the expected types.

So on the consuming side we keep using `StreamID`, right?

File: packages/stream-model-instance-handler/src/__tests__/model-instance-document-handler.test.ts [hunk 3]
```diff
@@ -347,6 +388,107 @@ describe('ModelInstanceDocumentHandler', () => {
...
+    const commitData = {
+      cid: FAKE_CID_1,
+      type: CommitType.GENESIS,
+      commit,
+    }
+    const streamState = await handler.applyCommit(commitData, context)
+    expect(streamState).toMatchSnapshot()
+  })
+
+  it('deterministic genesis commit cannot have content', async () => {
```
Comment (oed) [inline packages/stream-model-instance-handler/src/__tests__/model-instance-document-handler.test.ts:426]: 👍

**Review by oed — APPROVED** [review]
lgtm

File: packages/stream-model-instance/src/model-instance-document.ts (hunk 1 above)
Comment (stbrody) [reply packages/stream-model-instance/src/model-instance-document.ts:141]: @oed any opinion on `single()` vs `deterministic()`.  I think I very slightly lean towards `single`?  But it's very slight.

Comment (oed) [reply packages/stream-model-instance/src/model-instance-document.ts:141]: `single` seems nice

**Pull request merged**